src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/General/RoAssemblyName.cs

```diff
@@ -49,9 +50,9 @@ public RoAssemblyName(string? name, Version? version, string? cultureName, byte[
         // Equality - this compares every bit of data in the RuntimeAssemblyName which is acceptable for use as keys in a cache
         // where semantic duplication is permissible. This method is *not* meant to define ref->def binding rules or
         // assembly binding unification rules.
-        public bool Equals(RoAssemblyName other)
+        public bool Equals([AllowNull] RoAssemblyName other)
         {
-            if (Name != other.Name)
+            if (Name != other!.Name) // TODO2
```

Comment:
:memo: not sure what to do here

Reply:
why did we need to make it nullable?

Reply:
This is an internal type, and presumably it's not expecting Equals to be used with null, even though it implements an interface that allows it.  Seems like the right change is to remove this ! and ? and suppress the warning on the Equals method signature.

Reply:
To implement `IEquatable` you must be compatible with the signature `bool Equals([AllowNull] T other);`

Reply:
> To implement IEquatable you must be compatible with the signature bool Equals([AllowNull] T other);

I understand that.  So I'm saying use `#pragma warning disable whatever` to suppress the warning.